### PR TITLE
search: return communities a user is member of via search_user_communities

### LINKED ISSUE #503: Backend: Communities: search_user_communites() should only return communities for the given user
The ``CommunityService.search_user_communities()`` should only return communities that the given identity is a member of.

Essentially, we need to apply a query filter:

```
extra_filter=Q("terms", id=[com_uuid1, com_uuid2, ...])
```

Likely the easiest is to use the "CommunitMembers" generator to extra the list of communities from the identity, so it becomes something like:

```python
search_result = self._search(
    #...
    extra_filter=CommunityMembers().query_filter(identity=identity)
)
```

A tests should be added to show that it works.

Out of scope: Right now we don't have support for saying that an identity have access to **all** communities (e.g. super admin).

### PR DESCRIPTION
**Pull request by fenekku**
closes #503

### CODE REVIEW

File: invenio_communities/generators.py [hunk 1]
```diff
@@ -176,7 +176,7 @@ def query_filter(self, identity=None, **kwargs):
         # Gives access to all community members.
         return Q(
             "terms",
-            **{"id": self.communities(identity)}
+            **{"_id": self.communities(identity)}
```
Comment (lnielsen) [inline invenio_communities/generators.py:179]: Question: Is this a fix to a bug you've found?

**Review by lnielsen — COMMENTED** [review]
Thanks a lot! Just smaller comments from me.

**Comment by fenekku** [comment]
Didn't have time until today to look into this. Looking at this now.

File: invenio_communities/generators.py (hunk 1 above)
Comment (fenekku) [reply invenio_communities/generators.py:179]: Yes I suppose. `"id"` for a community is still the slug, but the CommunityRoleNeed stores the uuid, so the permission search didn't work without this change. What I've noticed was that a `uuid` (literally) field was also present in the document during tests, but I couldn't find its matching official mapping, so I went for `_id` which is guaranteed to be there. It is a little strange given that `_id` is the ES document id, but again I didn't see another official field where the uuid is kept.

**Comment by fenekku** [comment]
changes made.

File: tests/communities/test_services.py [hunk 2]
```diff
@@ -206,8 +212,72 @@ def test_update_featured(community_service, comm):
 
 def test_cleanup_pre_search(db, es_clear):
     """Cleanup database and elasticsearch for following tests.
-    
+
     Tests before do not depend on a clean db or es. Cleanup is also an
     expensive task so te tests will run faster without them.
     """
     pass
+
```
[the remark this answers is not in the document]
Comment (fenekku) [reply tests/communities/test_services.py:220]: done

**Comment by fenekku** [comment]
How can we improve this situation where a PR is made, but left hanging for weeks, so eventually other releases break its tests? 

Perhaps there is a systematic change we can do to help... :thinking:

**Comment by fenekku** [comment]
rebased

**Comment by lnielsen** [comment]
Thanks for updating. The rate of change has been so high on some of the repositories and with big far reaching changes, that even PRs made the same day would break. Definitely we can improve the situation, and we will discuss how, but it's also not unavoidable.

**Pull request merged**